tests/e2e-kubernetes/testsuites/cache.go

```diff
@@ -183,13 +182,9 @@ func (t *s3CSICacheTestSuite) DefineTests(driver storageframework.TestDriver, pa
 			mountOptions := append(baseMountOptions,
 				"allow-delete",
 				"allow-other",
-				fmt.Sprintf("uid=%d", *e2epod.GetDefaultNonRootUser()),
+				fmt.Sprintf("uid=%d", defaultNonRootUser),
 				fmt.Sprintf("gid=%d", defaultNonRootGroup))
-			podModifiers := append(basePodModifiers, func(pod *v1.Pod) {
-				pod.Spec.Containers[0].SecurityContext.RunAsUser = e2epod.GetDefaultNonRootUser()
-				pod.Spec.Containers[0].SecurityContext.RunAsGroup = ptr.To(defaultNonRootGroup)
-				pod.Spec.Containers[0].SecurityContext.RunAsNonRoot = ptr.To(true)
-			})
+			podModifiers := append([]func(*v1.Pod){podModifierNonRoot}, basePodModifiers...)
```

Comment:
Same here, could we add the file's namespace here so it's easier to search? Also, why does it need the `[]func(*v1.Pod){...}` bit?

Reply:
For local/unexported identifiers (i.e., the identifiers starts with a lowercase, see https://go.dev/tour/basics/3) that means they're coming from the same package and there is no importing mechanism for them.

This `append` block basically infers the type of slice, and `[]func(*v1.Pod){...}` is to make sure it infers the correct type.